### ISSUE TITLE
Build error with defaultLayout: The page component must export a React component for it to be valid

### ISSUE THREAD

**Issue by alexchantastic** [ChristopherBiscardi/gatsby-mdx#178]
**Describe the bug**
Using `defaultLayout` instead of defining a layout export in your `.mdx` files throws a build error:

```
The page component must export a React component for it to be valid
```

It seems that gatsby `v2.0.23+` throws an actual error while earlier versions will simply throw a warning and allow the build to progress.

**To Reproduce**
[gatsby-mdx minimal repro](https://github.com/alexchantastic/gatsby-mdx-minimal-repro/tree/missing_export)

**Comment by ChristopherBiscardi** [comment]
So this was introduced by [changing a `console.log` to a `panicOnBuild`](https://github.com/gatsbyjs/gatsby/pull/9031/files#diff-f61af2b51837961d211c7ea1a4b91837L294) in gatsby core and is going to be reverted by only [scoping that check to .jsx/tsx](https://github.com/gatsbyjs/gatsby/pull/9232) files.

Thanks for filing the issue @alexchantastic

**Comment by ChristopherBiscardi** [comment]
This fix was released in gatsby@2.0.27

**Comment by ChristopherBiscardi** [comment]
@OFranke that seems like a .ts issue unrelated to default layouts, but if you open a new issue with a [minimal repro](https://github.com/ChristopherBiscardi/gatsby-mdx-minimal-repro) I'm happy to take a look.
